modules/options/src/main/scala/scala/build/options/ScalaNativeOptions.scala

```diff
@@ -20,13 +38,27 @@ final case class ScalaNativeOptions(
   linkingDefaults: Option[Boolean] = None,
   compileOptions: List[String] = Nil,
   compileDefaults: Option[Boolean] = None,
-  embedResources: Option[Boolean] = None
+  embedResources: Option[Boolean] = None,
+  buildTargetStr: Option[String] = None
 ) {
 
   def finalVersion = version.map(_.trim).filter(_.nonEmpty).getOrElse(Constants.scalaNativeVersion)
 
   def numeralVersion = SNNumeralVersion.parse(finalVersion)
 
+  def target(): Option[ScalaNativeTarget] =
+    buildTargetStr.flatMap(ScalaNativeTarget.fromString)
+
+  private def targetCliOption(): List[String] =
+    import ScalaNativeTarget.*
+    val targ = target().map {
+      case Application    => "application"
+      case LibraryDynamic => "library-dynamic"
+      case LibraryStatic  => "library-static"
```

Comment:
Have you considered `dynamic-library` and `static-library`?

Reply:
These values actually have to match what Scala Native CLI uses: https://github.com/scala-native/scala-native-cli/blob/9d541e5c3a107b006cd61e7ece88df5cda7072e0/cli/src/main/scala/scala/scalanative/cli/options/NativeConfigOptions.scala#L40